core/src/main/java/org/apache/iceberg/FindFiles.java

```diff
@@ -206,7 +212,8 @@ public Builder inPartitions(PartitionSpec spec, List<StructLike> partitions) {
           .caseSensitive(caseSensitive)
           .entries();
 
-      return CloseableIterable.transform(entries, entry -> entry.file().copyWithoutStats());
+      return CloseableIterable.transform(entries,
+          entry -> includeColumnStats ? entry.file().copy() : entry.file().copyWithoutStats());
```

Comment:
Thanks @Flyangz for contributing,  but I don't think adding an option without any test case verification is the correct practice.  I think you can read this [PR](https://github.com/apache/iceberg/pull/206/files) to add the `includeColumnStats` in FindFiles API.

Reply:
Thanks reply. I have added the unit test, please check it out.